### PR TITLE
release: v1.9.1

### PR DESCRIPTION
**Pull request by claytonparnell**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by claytonparnell** [comment]
# Staleness Report: 1.9.1(gpu)

Package | Current Version in the Distribution image | Latest Relevant Version in Upstream
---|---|---
numpy|1.26.4|1.26.4
jinja2|3.1.4|3.1.4
pandas|2.1.4|2.1.4
altair|5.3.0|5.3.0
${\color{red}boto3}$|1.34.131|1.34.150
ipython|8.25.0|8.25.0
jupyter-lsp|2.2.5|2.2.5
amazon-sagemaker-jupyter-scheduler|3.1.2|3.1.2
amazon-sagemaker-sql-magic|0.1.3|0.1.3
jupyterlab-lsp|5.0.3|5.0.3
amazon_sagemaker_sql_editor|0.1.10|0.1.10
scipy|1.11.4|1.11.4
scikit-learn|1.4.2|1.4.2
pip|23.3.2|23.3.2
torchvision|0.15.2|0.15.2
autogluon|0.8.3|0.8.3
ipywidgets|8.1.3|8.1.3
notebook|7.1.3|7.1.3
aws-glue-sessions|1.0.6|1.0.6
conda|23.11.0|23.11.0
fastapi|0.110.3|0.110.3
jupyter-collaboration|1.1.0|1.1.0
jupyter-dash|0.4.2|0.4.2
jupyter-server-proxy|4.2.0|4.2.0
jupyterlab-git|0.50.1|0.50.1
keras|2.14.0|2.14.0
${\color{red}langchain-aws}$|0.1.6|0.1.12
matplotlib|3.8.4|3.8.4
nodejs|18.20.3|18.20.3
py-xgboost-gpu|1.7.6|1.7.6
thrift_sasl|0.4.3|0.4.3
pyhive|0.7.0|0.7.0
python-gssapi|1.8.3|1.8.3
python-lsp-server|1.11.0|1.11.0
pytorch-gpu|2.0.0|2.0.0
sagemaker-code-editor|1.1.0|1.1.0
sagemaker-headless-execution-driver|0.0.13|0.0.13
sagemaker-jupyterlab-emr-extension|0.1.9|0.1.9
sagemaker-jupyterlab-extension|0.3.2|0.3.2
sagemaker-kernel-wrapper|0.0.2|0.0.2
sagemaker-python-sdk|2.224.4|2.224.4
sagemaker-studio-analytics-extension|0.0.22|0.0.22
sasl|0.3.1|0.3.1
supervisor|4.2.5|4.2.5
tensorflow-gpu|2.14.0|2.14.0
uvicorn|0.30.3|0.30.3


# Staleness Report: 1.9.1(cpu)

Package | Current Version in the Distribution image | Latest Relevant Version in Upstream
---|---|---
numpy|1.26.4|1.26.4
jinja2|3.1.4|3.1.4
pytorch|2.0.0|2.0.0
pandas|2.1.4|2.1.4
altair|5.3.0|5.3.0
${\color{red}boto3}$|1.34.131|1.34.150
ipython|8.25.0|8.25.0
jupyter-lsp|2.2.5|2.2.5
${\color{red}jupyterlab}$|4.1.6|4.1.8
amazon-q-developer-jupyterlab-ext|3.1.0|3.1.0
${\color{red}langchain}$|0.1.9|0.1.20
jupyter-ai|2.18.1|2.18.1
amazon-sagemaker-jupyter-ai-q-developer|1.0.7|1.0.7
jupyter-scheduler|2.7.1|2.7.1
amazon-sagemaker-jupyter-scheduler|3.1.2|3.1.2
amazon-sagemaker-sql-magic|0.1.3|0.1.3
jupyterlab-lsp|5.0.3|5.0.3
amazon_sagemaker_sql_editor|0.1.10|0.1.10
scipy|1.11.4|1.11.4
scikit-learn|1.4.2|1.4.2
pip|23.3.2|23.3.2
torchvision|0.15.2|0.15.2
autogluon|0.8.3|0.8.3
ipywidgets|8.1.3|8.1.3
notebook|7.1.3|7.1.3
aws-glue-sessions|1.0.6|1.0.6
conda|23.11.0|23.11.0
fastapi|0.110.3|0.110.3
jupyter-collaboration|1.1.0|1.1.0
jupyter-dash|0.4.2|0.4.2
jupyter-server-proxy|4.2.0|4.2.0
jupyterlab-git|0.50.1|0.50.1
keras|2.15.0|2.15.0
${\color{red}langchain-aws}$|0.1.6|0.1.12
matplotlib|3.8.4|3.8.4
nodejs|18.20.3|18.20.3
py-xgboost-cpu|1.7.6|1.7.6
thrift_sasl|0.4.3|0.4.3
pyhive|0.7.0|0.7.0
python-gssapi|1.8.3|1.8.3
python-lsp-server|1.11.0|1.11.0
sagemaker-code-editor|1.1.0|1.1.0
sagemaker-headless-execution-driver|0.0.13|0.0.13
sagemaker-jupyterlab-emr-extension|0.1.9|0.1.9
sagemaker-jupyterlab-extension|0.3.2|0.3.2
sagemaker-kernel-wrapper|0.0.2|0.0.2
sagemaker-python-sdk|2.224.4|2.224.4
sagemaker-studio-analytics-extension|0.0.22|0.0.22
sasl|0.3.1|0.3.1
supervisor|4.2.5|4.2.5
tensorflow|2.15.0|2.15.0
uvicorn|0.30.3|0.30.

**Comment by claytonparnell** [comment]
Failed tests
```
============================================================================================== short test summary info ===============================================================================================
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[keras.test.Dockerfile-required_packages0] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[scipy.test.Dockerfile-required_packages4] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[sm-python-sdk.test.Dockerfile-required_packages8] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[pytorch.examples.Dockerfile-required_packages9] - assert 1 == 0
```
Confirmed that these failed with the issues tracked
* keras - https://github.com/aws/sagemaker-distribution/issues/413
* scipy - https://github.com/aws/sagemaker-distribution/issues/30 
* sm-python-sdk - https://github.com/aws/sagemaker-distribution/issues/316

pytorch succeeded when reran

File: build_artifacts/v1/v1.9/v1.9.1/gpu.env.in [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+conda-forge::sagemaker-code-editor[version='>=1.1.0,<1.2.0']
+conda-forge::amazon-sagemaker-jupyter-ai-q-developer[version='>=1.0.4,<1.1.0']
+conda-forge::amazon-q-developer-jupyterlab-ext[version='>=3.1.0,<3.2.0']
+conda-forge::amazon_sagemaker_sql_editor[version='>=0.1.9,<0.2.0']
+conda-forge::amazon-sagemaker-sql-magic[version='>=0.1.3,<0.2.0']
+conda-forge::langchain[version='>=0.1.9,<0.2.0']
+conda-forge::fastapi[version='>=0.110.3,<0.111.0']
+conda-forge::uvicorn[version='>=0.30.1,<0.31.0']
+conda-forge::pytorch-gpu[version='>=2.0.0,<2.1.0']
+conda-forge::tensorflow-gpu[version='>=2.14.0,<2.15.0']
```
Comment (TRNWWZ) [inline build_artifacts/v1/v1.9/v1.9.1/gpu.env.in:13]: Is this introduced recently?

**Comment by TRNWWZ** [comment]
May not related to this PR. But why in staleness report, `pytorch` and `pytorch-gpu` Latest Relevant Version in Upstream is only 2.0.0?

**Comment by claytonparnell** [comment]
> May not related to this PR. But why in staleness report, `pytorch` and `pytorch-gpu` Latest Relevant Version in Upstream is only 2.0.0?

Since this is a patch release, the latest relevant version would be latest patch from v2.0. So since there's no 2.0.1, 2.0.0 is latest "relevant" version

File: build_artifacts/v1/v1.9/v1.9.1/gpu.env.in (hunk 1 above)
Comment (claytonparnell) [reply build_artifacts/v1/v1.9/v1.9.1/gpu.env.in:13]: Yeah in 1.9.0 we switched to fix the gpu issue in 1.8. From 2.0 we switch back to standard tensorflow

**Comment by TRNWWZ** [comment]
Do we need the change to remove conda default channel and use conda-forge only in Docker file?

**Comment by claytonparnell** [comment]
> Do we need the change to remove conda default channel and use conda-forge only in Docker file?

No, we're doing that change from 2.0. For v1 since it's breaking change we are keeping it the same

**Review by TRNWWZ — APPROVED** [review]

**Review by JunLyu — APPROVED** [review]
